### PR TITLE
chore: Ruby agent: 8.7.0 Release

### PR DESCRIPTION
**Pull request by kaylareopelle**
## Give us some context

* What problems does this PR solve?
- Updates configuration for changes included in the 8.7.0 release
- Adds notes about deprecated gem instrumentation
- Expands Custom instrumentation article to include details about Async instrumentation and RubyDoc API documentation

### CODE REVIEW

**Comment by barbnewrelic** [comment]
@kaylareopelle  I see this is set as a draft. Is there an ETA for v8.7.0? Will there be a separate PR for that release note, or will it be added to this one? Thanks for confirming!

**Comment by rhetoric101** [comment]
Temporarily closing to see if a reopening will trigger a successful build.

File: src/content/docs/apm/agents/ruby-agent/api-guides/ruby-custom-instrumentation.mdx [hunk 1]
```diff
@@ -15,7 +15,7 @@ redirects:
   - /docs/agents/ruby-agent/customization
 ---
 
-The New Relic Ruby agent automatically collects many metrics. It also includes an API you can use to collect additional metrics about your application. If you see large **Application Code** segments in transaction trace details, custom instrumentation can give a more complete picture of what is going on in your application.
+The New Relic Ruby agent automatically collects many metrics. It also includes an API you can use to collect additional metrics about your application. If you see large **Application Code** segments in transaction trace details, custom instrumentation can give a more complete picture of what is going on in your application. For details about all available API methods, please see our [api documentation](https://www.rubydoc.info/gems/newrelic_rpm).
```
Comment (paperclypse) [inline src/content/docs/apm/agents/ruby-agent/api-guides/ruby-custom-instrumentation.mdx:18]:
```suggestion
The New Relic Ruby agent automatically collects many metrics. It also includes an API you can use to collect additional metrics about your application. If you see large **Application Code** segments in transaction trace details, custom instrumentation can give a more complete picture of what is going on in your application. For details about all available API methods, please see our [Ruby agent API documentation](https://www.rubydoc.info/gems/newrelic_rpm).
```

Tweaking the link text to make it a little more accessible.

File: src/content/docs/apm/agents/ruby-agent/api-guides/ruby-custom-instrumentation.mdx [hunk 2]
```diff
@@ -63,7 +63,7 @@ class Foo
 end
 ```
 
-`add_method_tracer` takes an optional metric name and a hash of options. For more information, see [add_method_tracer in the New Relic RubyDoc](https://rubydoc.info/github/newrelic/newrelic-ruby-agent/NewRelic/Agent/MethodTracer/ClassMethods#add_method_tracer-instance_method) .
+`add_method_tracer` takes an optional metric name and a hash of options. For more information, see [add_method_tracer in the New Relic RubyDoc](https://www.rubydoc.info/gems/newrelic_rpm/NewRelic/Agent/MethodTracer/ClassMethods#add_method_tracer-instance_method) .
```
Comment (paperclypse) [inline src/content/docs/apm/agents/ruby-agent/api-guides/ruby-custom-instrumentation.mdx:66]:
```suggestion
`add_method_tracer` takes an optional metric name and a hash of options. For more information, see [add_method_tracer in the New Relic RubyDoc](https://www.rubydoc.info/gems/newrelic_rpm/NewRelic/Agent/MethodTracer/ClassMethods#add_method_tracer-instance_method).
```

File: src/content/docs/apm/agents/ruby-agent/api-guides/ruby-custom-instrumentation.mdx [hunk 3]
```diff
@@ -104,7 +104,7 @@ def slow_action
 end
 ```
 
-For more information, see [add_method_tracer in the New Relic RubyDoc](https://www.rubydoc.info/github/newrelic/newrelic-ruby-agent/NewRelic/Agent/MethodTracer#trace_execution_scoped-instance_method) .
+For more information, see [add_method_tracer in the New Relic RubyDoc](https://www.rubydoc.info/gems/newrelic_rpm/NewRelic/Agent/MethodTracer#trace_execution_scoped-instance_method) .
```
Comment (paperclypse) [inline src/content/docs/apm/agents/ruby-agent/api-guides/ruby-custom-instrumentation.mdx:107]:
```suggestion
For more information, see [add_method_tracer in the New Relic RubyDoc](https://www.rubydoc.info/gems/newrelic_rpm/NewRelic/Agent/MethodTracer#trace_execution_scoped-instance_method).
```

File: src/content/docs/apm/agents/ruby-agent/api-guides/ruby-custom-instrumentation.mdx [hunk 4]
```diff
@@ -237,4 +237,17 @@ class MyEventWatcher
...
+
+To support applications that want to control which threads are instrumented by New Relic, a class is available in 8.7.0 and above, `NewRelic::TracedThread`, that will create a thread that includes New Relic instrumentation. 
+Use this class as a replacement for the native Thread class. Please see our [API documentation](https://www.rubydoc.info/gems/newrelic_rpm/NewRelic/TracedThread) for more details. 
+```
+  NewRelic::TracedThread.new do 
+    # custom code to run inside of the thread
+  end
+
+```
+To have the agent automatically instrument all threads, please see the configuration `instrumentation.thread.tracing` [documentation](https://docs.newrelic.com/docs/apm/agents/ruby-agent/configuration/ruby-agent-configuration/#instrumentation).
```
Comment (paperclypse) [inline src/content/docs/apm/agents/ruby-agent/api-guides/ruby-custom-instrumentation.mdx:252]:
```suggestion
In 8.7.0 and higher, use the class `NewRelic::TracedThread` in applications to create threads that are instrumented by New Relic.

Use this class as a replacement for the native `Thread` class. See our [TracedThread API documentation](https://www.rubydoc.info/gems/newrelic_rpm/NewRelic/TracedThread) for more detail. 

```
  NewRelic::TracedThread.new do 
    # custom code to run inside of the thread
  end

```

To automatically instrument all threads, see the [configuration `instrumentation.thread.tracing` documentation](https://docs.newrelic.com/docs/apm/agents/ruby-agent/configuration/ruby-agent-configuration/#instrumentation).
```

File: src/content/docs/apm/agents/ruby-agent/getting-started/ruby-agent-requirements-supported-frameworks.mdx [hunk 5]
```diff
@@ -170,7 +170,7 @@ Web servers supported by the Ruby agent include:
       <td/>
 
       <td>
-        * 4.5.0
+        * 4.5.0 - deprecated
```
Comment (paperclypse) [inline src/content/docs/apm/agents/ruby-agent/getting-started/ruby-agent-requirements-supported-frameworks.mdx:173]:
```suggestion
        * 4.5.0 - Deprecated
```

**Review by paperclypse — APPROVED** [review]
@kaylareopelle Looks good! This was a big PR. Thanks for the work and apologies for the technical difficulties!

**Comment by kaylareopelle** [comment]
Thanks for making the fixes and merging, @paperclypse! I'm sorry I missed your comments as they rolled out!

**Pull request merged**